### PR TITLE
chore: replace close icon with `secondary_icon` variant in Alert, Modal, Popover, Toast

### PR DESCRIPTION
**Pull request by nkrantz**
[Ticket](https://issues.corp.twilio.com/browse/DSYS-2269)

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [ ] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

**Review by TheSisb — APPROVED** [review]

**Review by gloriliale — APPROVED** [review]
Checked em all out and they look great!

File: packages/paste-core/components/alert/src/index.tsx [hunk 1]
```diff
@@ -112,7 +112,7 @@ const Alert = React.forwardRef<HTMLDivElement, AlertProps>(
           <MediaBody as="div">{children}</MediaBody>
           {onDismiss && typeof onDismiss === 'function' && (
             <MediaFigure align="end" spacing="space60">
-              <Button onClick={onDismiss} variant="link" size="reset" element={`${element}_DISMISS_BUTTON`}>
+              <Button onClick={onDismiss} variant="secondary_icon" size="reset" element={`${element}_DISMISS_BUTTON`}>
```
Comment (shleewhite) [inline packages/paste-core/components/alert/src/index.tsx:115]: Can you update the close icon's title to be "Dismiss alert"? would make the language more concise, the word this doesnt really do anything

File: packages/paste-core/components/toast/src/Toast.tsx [hunk 2]
```diff
@@ -89,7 +89,7 @@ const Toast = React.forwardRef<HTMLDivElement, ToastProps>(
             <MediaFigure align="end" spacing="space40">
               <Button
                 onClick={onDismiss}
-                variant="link"
+                variant="secondary_icon"
```
Comment (shleewhite) [inline packages/paste-core/components/toast/src/Toast.tsx:92]: Same as the alert, can the title of the close icon be "Dismiss toast".

**Review by shleewhite — APPROVED** [review]
Super small comments about changing the close icon titles.

**Comment by nkrantz** [comment]
@TheSisb @shleewhite pushed up with your changes, just want to confirm with you that they're all good before merging

**Pull request merged**